packages/scanner/src/appMapIndex.ts

```diff
@@ -3,7 +3,7 @@ import { QueryAST } from './types';
 import LRUCache from 'lru-cache';
 
 const NormalizedSQLBySQLString = new LRUCache<string, string>({ max: 10000 });
-const ASTBySQLString = new LRUCache<string, QueryAST>({ max: 1000 });
+const ASTBySQLString = new LRUCache<string, QueryAST | 'parse-error'>({ max: 1000 });
```

Comment:
Because the LRUCache could contain `null` as a value, unparseable SQL would be parsed over and over.